### ISSUE TITLE
Flaky TestIdentifyPushOnAddrChange

### ISSUE THREAD

**Issue by Stebalien** [libp2p/go-libp2p#950]
```
--- FAIL: TestIdentifyPushOnAddrChange (0.03s)
    id_test.go:591: 
        	Error Trace:	id_test.go:591
        	Error:      	"[]" should have 1 item(s), but has 0
        	Test:       	TestIdentifyPushOnAddrChange
```

So, it looks like we're finishing the call to `Connect` before the other side sees the connection. However, that should be impossible at this point given that:

1. We wait for identify.
2. The other side won't respond to the identify request until after it has recorded the new connection.

So I'm not sure what's going on here.

**Comment by Stebalien** [comment]
@aarshkshah1992 pointed out that we're using the blank host here, which _doesn't_ have these guarantees.

**Comment by aarshkshah1992** [comment]
@Stebalien PR at https://github.com/libp2p/go-libp2p/pull/952.

**Comment by aarshkshah1992** [comment]
Fixed in #952 .
